internal/provisioner/plain/main.go

```diff
@@ -119,6 +120,7 @@ func main() {
 			SelectorsByObject: cache.SelectorsByObject{
 				&rukpakv1alpha1.BundleInstance{}: {},
 				&rukpakv1alpha1.Bundle{}:         {},
+				&corev1.Secret{}:                 {},
```

Comment:
@exdx The Get (sigs.k8s.io/controller-runtime/pkg/client) didn't work without it. Am I missing something to make it work?
```
       secret := &corev1.Secret{}
	err := r.Get(ctx, client.ObjectKey{Namespace: r.PodNamespace, Name: bundle.Spec.Source.Git.SecretName}, secret)
```
Should I change it to the client in `k8s.io/client-go/kubernetes`?
Thanks!

Reply:
Hmm that's interesting, I didn't expect that, but admittedly I don't know that much about the c/r client. I suppose maybe the c/r client only talks to its cache directly by design? 

I'm not sure about introducing the client-go client to the controller here. Maybe @timflannagan could give us a suggestion?

Reply:
Ideally, what we end up with is our cache contains secrets such that:
    - We can get/list/watch ALL secrets in the `rukpak-system` namespace
    - We can only get/list watch secrets with the rukpak selector labels in other namespaces.

Off the top of my head, I don't know whether there's a simple way to accomplish that with a single cache. We basically need the informer to have:
`metadata.namespace == "rukpak-system" || rukpakSelector.Match(metadata.labels)`

In the meantime, we could use `mgr.GetAPIReader` instead of `mgr.GetClient`. We'd bypass the cache with that which is:
- Good: we don't have to watch all secrets in the cluster
- Bad: we have to hit the API server during every reconcile of a bundle that contains a secret ref.

I'd propose that we go with the "Bad" alternative for now. And make a TODO to look into optimizing the way our cache is constructed.

Reply:
That would be better -- I agree that the "bad" alternative is good enough for now and we can potentially revisit later if we run into scaling issues.